addons/api/addon/serializers/host-set.js

```diff
@@ -19,10 +23,17 @@ export default class HostSetSerializer extends ApplicationSerializer {
    * @return {object}
    */
   serialize(snapshot) {
-    let serialized = super.serialize(...arguments);
     const hostIDs = snapshot?.adapterOptions?.hostIDs;
-    if (hostIDs) serialized = this.serializeWithHostIDs(snapshot, hostIDs);
-    return serialized;
+    if (hostIDs) {
+      return this.serializeWithHostIDs(snapshot, hostIDs);
+    } else {
+      switch (snapshot.record.compositeType) {
```

Comment:
A switch for just two cases might be killer, but I assume in the future we will have more cases. Happy to refactor for a simpler `if else` if you think we will not have more cases in the short-term.